content/articles/the-complete-roadmap-to-blockchain-development/index.md

```diff
@@ -0,0 +1,116 @@
+
+ For example, this is how you write `Hello World` in solidity
+ ```js
+ // My First Smart Contract 
+pragma solidity >=0.5.0 <0.7.0;
+contract HelloWorld {
+    function get()public pure returns (string memory){
+        return 'Hello Contracts';
+    }
+}
+ ```
+ I hope you can note the similarities to Javascript i.e the functions, the methods, the curly braces, etc. 
+
+ A great way to experiment with solidity is to use an online IDE called Remix. With Remix You just load the [website](https://remix.ethereum.org/), start coding, and run your first smart contract. [Here](https://www.tutorialspoint.com/solidity/index.htm) is a reference in case you want to get started with programming in solidity. However, in real-life projects, we need something more robust and thus most developers use Truffle which is a popular framework for smart contracts. "Truffle" is a command-line tool written in NodeJs and it can easily be installed on Windows, Mac OS, and Linux using the package manager npm.
+
+ Truffle comes with a  local ethereum network called Ganache that we use for testing our smart contract during development. With Ganache you can deploy your smart contract to an ethereum network completely separated from the real ethereum network called "mainnet".In this fake network, you can have an infinitely high amount of fake ether which means you can do as many transactions as you want and lose all the ether without real implications to your finances. This is to help you test your smart contract before deploying it to the real ethereum network.
+
+ You may also decide to use "Public testnets of ethereum" which are ethereum networks that are more realistic to the mainnet network as compared to Ganache. These public testnets are a little difficult to use since you don't control the network on your own. Herein you cannot generate infinite ether as in Ganache and thus you have to be prudent with the ether that you have. To generate ether on public testnets you would need to use tools called Faucets which usually fail sometimes. The public testnets are however a good way to get close to the real-world experience and put your smart contract to the test.
+
+ In the public testnets and the mainnet, you would need to make use of a service called Infura. Infura is an API that runs ethereum clients for you since it is not easy to run clients on your own. In addition to that, the way you send transactions in Ganache is quite different compared to how transactions take place in public testnets and mainnet. The learning curve increases exponentially with practice.
+
+ Finally, you need to test your smart contract appropriately before deploying it since as I mentioned earlier, the code is immutable. The testing is easily achieved using the truffle framework. 
```

Comment:
```suggestion
Finally, you need to test your smart contract appropriately before deploying it since as mentioned earlier, the code is immutable. The testing is easily achieved using the truffle framework.